src/fmm.c

```diff
@@ -1093,25 +1093,24 @@ void *fmm_allocate_doorbell(uint32_t gpu_id, uint64_t MemorySizeInBytes,
 static void *fmm_allocate_host_cpu(uint64_t MemorySizeInBytes,
 				HsaMemFlags flags)
 {
-	int err;
-	HSAuint64 page_size;
 	void *mem = NULL;
 	vm_object_t *vm_obj;
+	int mmap_prot = PROT_READ | PROT_WRITE;
 
-	page_size = PageSizeFromFlags(flags.ui32.PageSize);
-	err = posix_memalign(&mem, page_size, MemorySizeInBytes);
-	if (err != 0)
+	/* mmap will return a pointer with alignment equal to
+	 * sysconf(_SC_PAGESIZE).  We have stored this value in PAGE_SIZE.
+	 * We want to return a pointer aligned to
+	 * PageSizeFromFlags(flags.ui32.PageSize), so make sure this value
+	 * matches PAGE_SIZE.
+	 */
+	if ((int)PageSizeFromFlags(flags.ui32.PageSize) != PAGE_SIZE)
```

Comment:
I'm not sure if anyone is using it, but we should at least support HSA_PAGE_SIZE_2MB too. I think this is just a hint, and doesn't really require a specific alignment. I would just ignore the page size flag.